### PR TITLE
Check if BEEFY authority is in current set

### PR DESCRIPTION
**Pull request by seunlanlege**
closes paritytech/grandpa-bridge-gadget#100

### CODE REVIEW

File: client/beefy/src/round.rs [hunk 1]
```diff
@@ -82,7 +82,10 @@ where
 	}
 
 	pub(crate) fn add_vote(&mut self, round: (H, N), vote: (Public, Signature)) -> bool {
-		self.rounds.entry(round).or_default().add_vote(vote)
+		if let Some(_) = self.validator_set.validators.iter().find(|id| vote.0 == **id) {
+			return self.rounds.entry(round).or_default().add_vote(vote);
+		}
+		false
```
Comment (bkchr) [inline client/beefy/src/round.rs:88]:
```suggestion
		if self.validator_set.validators.iter().any(|id| vote.0 == **id) {
			self.rounds.entry(round).or_default().add_vote(vote)
		} else {
			false
		}
```

File: client/beefy/src/round.rs [hunk 2]
```diff
@@ -82,7 +82,11 @@ where
 	}
 
 	pub(crate) fn add_vote(&mut self, round: (H, N), vote: (Public, Signature)) -> bool {
-		self.rounds.entry(round).or_default().add_vote(vote)
+		if self.validator_set.validators.iter().any(|id| vote.0 == **id) {
```
Comment (bkchr) [inline client/beefy/src/round.rs:85]:
```suggestion
		if self.validator_set.validators.iter().any(|id| vote.0 == *id) {
```

File: client/beefy/src/round.rs [hunk 3]
```diff
@@ -36,7 +36,7 @@ impl RoundTracker {
 	fn add_vote(&mut self, vote: (Public, Signature)) -> bool {
 		// this needs to handle equivocations in the future
 		if self.votes.contains(&vote) {
-			return false
+			return false;
```
Comment (adoerr) [inline client/beefy/src/round.rs:39]:
```suggestion
			false
```

Comment (adoerr) [reply client/beefy/src/round.rs:39]: Looks like `cargo fmt` doesn't like the semicolon here

**Comment by seunlanlege** [comment]
@adoerr looks good?

**Review by adoerr — APPROVED** [review]

**Review by bkchr — APPROVED** [review]

**Comment by tomusdrw** [comment]
A bit shame that such an important piece of code was accepted without a single test case.

**Comment by seunlanlege** [comment]
🥲

**Comment by adoerr** [comment]
I'll add some

**Pull request merged**